app/app.rb

```diff
@@ -50,12 +52,19 @@ class App < Padrino::Application
     end
 
     get :index, :with => :endpoint do
-      content_type :json
+      endpoint = params.delete('endpoint')
+
+      format = File.extname(endpoint)
+      endpoint = File.basename(endpoint, format)
+      format[0] = ''
+      format = 'json' if format.empty?
+      content_type format.to_sym
```

Comment:
@cantino I don't believe it would be a security risk, since they can't specify the data, but now that you mention it, we should validate that it is one of the supported types.  thanks!